burn-core/src/nn/conv/conv1d.rs

```diff
@@ -62,18 +67,33 @@ pub struct Conv1d<B: Backend> {
 impl Conv1dConfig {
     /// Initialize a new [conv1d](Conv1d) module.
     pub fn init<B: Backend>(&self) -> Conv1d<B> {
-        let shape = [self.channels_out, self.channels_in, self.kernel_size];
+        let shape = [
+            self.channels_out,
+            self.channels_in / self.groups,
+            self.kernel_size,
+        ];
         let fan_in: usize = self.channels_in * self.kernel_size;
```

Comment:
TL;DR: divide fan_in by self.groups and revert initialization to what we had before. 

The former uniform Kaiming initialization
Was equivalent to
gain * Uniform {-k, k} with k = sqrt( 3 / fan_in) and gain = sqrt( 1 / 3 )
Which is therefore equivalent to
Uniform {-k, k} with k = sqrt( 1 / fan_in ) 

You have replaced it with
Uniform {-k, k} with k = sqrt( groups / fan_in )
With fan_in unchanged (i.e. still equal to channels_in * kernel_size). 

If you divide fan_in by self.groups, then you will get 
sqrt( 1 / new_fan_in ) = sqrt( 1 / (old_fan_in / groups) ) = sqrt( groups / old_fan_in ), leading to exactly the same thing you had. 
Therefore you can revert to the Kaiming initialization we had before. It will simply use a fan_in that is divided by the number of groups instead